firmware/application/apps/ui_level.cpp

```diff
@@ -287,6 +286,7 @@ size_t LevelView::change_mode(freqman_index_t new_mod) {
                 receiver_model.set_sampling_rate(actual_sampling_rate);
                 receiver_model.set_baseband_bandwidth(filter_bandwidth_for_sampling_rate(actual_sampling_rate));
             };
+            field_bw.set_by_value(0);
```

Comment:
In SPEC we need some calculation before setting the bandwidth, and there is no 'set_spec_configuration blah' so you have to call the set_by_value after the  on_change callback is defined